logpush-to-bigquery/index.js

```diff
@@ -26,17 +22,75 @@ async function gcsbq (file, context) {
     ignoreUnknownValues: true
   }
 
-  const addToTable = async (tableId) => {
+  const addToTable = async tableId => {
     const dataset = await bigquery.dataset(datasetId).get({ autoCreate: true })
     const table = await dataset[0].table(tableId).get({ autoCreate: true })
-    return table[0].load(filename, metadata)
+    return table[0].createLoadJob(files, metadata)
   }
 
   try {
-    await addToTable(tableId)
+    return addToTable(tableId)
   } catch (e) {
     console.log(e)
   }
 }
 
-module.exports.gcsbq = gcsbq
+async function writeLog (logData) {
+  logData = logData.join(',\n')
+
+  const metadata = {
+    resource: { type: 'global' },
+    severity: 'INFO'
+  }
+
+  const entry = log.entry(metadata, logData)
+  await log.write(entry)
+  console.log(`Logged: ${logData}`)
+}
+
+module.exports.runLoadJob = async function (message, context) {
+  if (!context) {
+    context = {}
+    context.timestamp = new Date().toISOString()
+  }
+  context.timestamp = DateTime.fromISO(context.timestamp)
+
+  const loadJobDeadline = context.timestamp
+    .setZone('GMT')
+    .minus({ minutes: 3 })
+    .startOf('minute')
+
+  const [deadlineDate, deadlineTime] = [
+    loadJobDeadline.toFormat('yyyyMMdd'),
+    loadJobDeadline.toFormat(`yyyyMMdd'T'hhmm`)
+  ]
+
+  let arrayOfLogpushLogs = []
+  let stackdriverEntry = []
+
+  try {
+    bucket
+      .getFilesStream({
+        autoPaginate: false,
+        maxResults: 5000,
+        prefix: `${process.env.DIRECTORY}${deadlineDate}/${deadlineTime}`
```

Comment:
A better approach would be to use the 'second' date (the flush time) rather than the 'first' (start of buffer).  This would do a better job of capturing all data when there is some sort of lag in the system that causes the flush time to be extended.

I think you can use a wildcard to replace the first deadlineDate/deadlineTime - but I had some trouble getting this to work with a quick test.

Reply:
Let's discuss this in the thread below.